src/components/forms/Select/Select.styles.tsx

```diff
@@ -485,6 +485,7 @@ export const Option: ComponentType<OptionProps<OptionType, boolean>> = (
       <components.Option {...props}>
         <Inline gap={SpaceSizes.sm}>
           <Checkbox
+            checkboxId="select"
```

Comment:
@Mnau-mnau I guess this will be problem, the checkboxId has to be unique so probably something like 
```
checkboxId={`select-${label}`}